packages/charts/src/charts/line/line-chart.utils.ts

```diff
@@ -0,0 +1,119 @@
+import { scaleLinear, scaleUtc } from 'd3-scale';
+import { line as lineShape } from 'd3-shape';
+import { calculateRange, calculateScaleDomain } from '../../utils';
+import { Dimension, Margins } from '../../types';
+
+type Options = {
+  data: any[];
+  keys: string[];
+  labelSelector: string;
+  dimension: Dimension;
+  margins: Margins;
+  minValue?: number | 'auto';
+  maxValue?: number | 'auto';
+  colors: string[];
+  markRadius: number;
+  strokeWidth: number;
+};
+
+export type Lines = {
+  key: string;
+  line: Line;
+  marks: Mark[];
+  color: string;
+  markRadius: number;
+  strokeWidth: number;
+};
+
+export type Line = {
+  key: string;
+  d: string;
+};
+
+export type Mark = {
+  key: string;
+  x: number;
+  y: number;
+};
+
+export const sortDates = (data: any[], labelSelector: string) =>
+  data.length &&
+  data.sort(
+    (a, b) =>
+      (new Date(a[labelSelector]) as any) - (new Date(b[labelSelector]) as any)
```

Comment:
`any` -> `number ` ?

Reply:
Number will not work, because there is a Date type, and Date type will not work because there is an error:
`...-hand side of an arithmetic operation must be of type 'any', 'number', 'bigint' or an enum type.`